frontend/src/createStream/__tests__/StreamingTable.test.js

```diff
@@ -0,0 +1,239 @@
+import React from 'react';
+import axios from "axios";
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+import StreamingTable from "../StreamingTable";
+
+jest.mock('axios');
+jest.spyOn(global.console,"log");
+
+class DummyData {
+
+    getSenders() {
+        return ["A", "B", "C"];
+    }
+
+    getReceivers() {
+        return ["X", "Y", "Z"];
+    }
+
+    preventDefault(){
+        return;
+    }
+
+}
+describe("<StreamingTable/>", () => {
+    let wrapper;
+
+    beforeEach(() => {
+
+        wrapper = Enzyme.shallow(<StreamingTable dataSource={new DummyData()} />)
```

Comment:
smart answer: Enzyme allows us to constrain the testing to the unit itself instead of indirectly testing children behaviours (Read more here: https://enzymejs.github.io/enzyme/docs/api/shallow.html )

Real answer: It was in every example I saw for testing states and I didn't think twice about it. Also it can read the state of something which is good because I'm not sure how to do that otherwise